client/my-sites/stats/site.jsx

```diff
@@ -18,88 +19,61 @@ import ChartTabs from './stats-chart-tabs';
 import StatsModule from './stats-module';
 import statsStrings from './stats-strings';
 import titlecase from 'to-title-case';
-import analytics from 'lib/analytics';
 import StatsFirstView from './stats-first-view';
 import config from 'config';
-
-const debug = debugFactory( 'calypso:stats:site' );
-
-module.exports = React.createClass( {
-	displayName: 'StatsSite',
-
-	getInitialState: function() {
-		const scrollPosition = this.props.context.state.scrollPosition || 0;
-
-		return {
-			date: this.props.date,
-			chartDate: this.props.date,
+import { getSelectedSiteId, getSelectedSiteSlug } from 'state/ui/selectors';
+import { getSiteOption, isJetpackSite } from 'state/sites/selectors';
+import { recordGoogleEvent } from 'state/analytics/actions';
+
+class StatsSite extends Component {
+	constructor( props ) {
+		super( props );
+		this.state = {
 			chartTab: this.props.chartTab,
-			tabSwitched: false,
-			period: this.props.period.period,
-			scrollPosition: scrollPosition
+			tabSwitched: false
 		};
-	},
-
-	componentWillReceiveProps: function( nextProps ) {
-		const newDate = this.moment( nextProps.date );
-		const newState = {
-			date: newDate,
-			chartDate: newDate
-		};
-
-		if ( ! this.state.tabSwitched || ( this.state.period !== nextProps.period.period ) ) {
-			newState.chartTab = nextProps.chartTab;
-			newState.period = nextProps.period.period;
-			newState.tabSwitched = true;
-		}
-
-		this.setState( newState );
-	},
+	}
 
-	scrollTop: function() {
-		if ( window.pageYOffset ) {
-			return window.pageYOffset;
+	componentWillReceiveProps( nextProps ) {
+		if ( ! this.state.tabSwitched && this.state.chartTab !== nextProps.chartTab ) {
+			this.setState( {
+				tabSwitched: true,
+				chartTab: nextProps.chartTab
+			} );
 		}
-		return document.documentElement.clientHeight ? document.documentElement.scrollTop : document.body.scrollTop;
-	},
-
-	componentDidMount: function() {
-		const scrollPosition = this.state.scrollPosition;
+	}
 
-		setTimeout( function() {
-			window.scrollTo( 0, scrollPosition );
+	componentDidMount() {
+		setTimeout( () => {
+			window.scrollTo( 0, 0 );
```

Comment:
Right! I think it does not have any impact on the sticky nav since the component is not unmounted I think, but yeah, I don't think this is useful, I hesitated.